### ISSUE TITLE
Mavlink v1: function mavlink_parse_char can "skip" ~264 bytes of data

### ISSUE THREAD

**Issue by bin-i** [ArduPilot/pymavlink#387]
I came across with this situation when a new message was added to the transmitter, but not added to the receiver. In this case the "mavlink_frame_char" function returns MAVLINK_FRAMING_BAD_CRC code and sometimes this message can end with 0xFE (CRC2 byte). 

```
| Start-of-frame | -/- | CRC1 | CRC2      | Start-of-frame  | -/----
| 0xFE           | -/- | -/-  | 0xFE      | 0xFE            | -/----
| Unknown message or message with bad CRC | Next messages   | -/----
```

In this case the "mavlink_parse_char" function sets status->parse_state to MAVLINK_PARSE_STATE_GOT_STX at CRC2 position, and reads Start-of-frame as Payload-length (254 bytes).

In other words when message with bad CRC ends with 0xFE, "mavlink_frame_char" function will confuse the end of the previous message with the beginning of the next message. And the beginning of the next will be interpreted as the length of the message.

**Comment by magicrub** [comment]
Interesting scenario!
